### ISSUE TITLE
Warn about [[ $foo == "*keyword*" ]]

### ISSUE THREAD

**Issue by guettli** [koalaman/shellcheck#2799]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

foo="foo Ready bar"
if [[ $foo == "*Ready*" ]]; then
    echo "Ready"
fi

```

#### Here's what shellcheck currently says:

no issues found

#### Here's what I wanted or expected to see:

I would like to get a warning, since this does not work as expected. You need `*Ready*` for pattern matching (without quotes).

**Comment by kurahaupo** [comment]
How is ShellCheck supposed to guess that you didn't want to check for the literal text `*Ready*`? (Perhaps you're checking a variable that holds patterns to see if it contains a particular pattern?)

You would expect `rm "*.txt"` to remove ONE (oddly named) file. Likewise you would expect `[[ $x = "*.txt" ]]` to tell you whether `x` contains that filename.

**Comment by kurahaupo** [comment]
We don't need ShellCheck to warn us about things that are *always* broken, such as extra quotes. We do need ShellCheck to warn us about things that are *sometimes* broken, such as missing quotes.

**Comment by guettli** [comment]
ok, thank you for your clear feedback.
